todo_app/tests/test_app.py

```diff
@@ -0,0 +1,131 @@
+    trello_token = os.environ.get('TRELLO_TOKEN')
+
+    if url == f'https://api.trello.com/1/boards/{test_board_id}/lists':
+        fake_response_data = [{
+            'id': '123abc',
+            'name': 'To Do',
+            'cards': [{'id': '456', 'name': 'Test card'}]
+        }]
+        return StubResponse(fake_response_data)
+    
+    elif url == f'https://api.trello.com/1/board/{test_board_id}/cards?key={trello_key}&token={trello_token}':
+        fake_response_data = [{
+            'id': '123abc',
+            'name': 'TestCard1',
+            'idList': '64faketodo1234e86c4c2ff3',
+            'desc': "Test desc",
+            'due': None},
+            {'id': '456def',
+            'name': 'TestCard2',
+            'idList': '64fakedone999985b9aac33',
+            'desc': "Test desc 2",
+            'due': '2023-09-05T09:59:00.000Z'
+        }]
+        return StubResponse(fake_response_data)
+    
+    elif url == f'https://api.trello.com/1/card/{test_card_id}?key={trello_key}&token={trello_token}':
+        fake_response_data = [{
+            'id': '123abc',
+            'name': 'TestCard1',
+            'idList': '64faketodo1234e86c4c2ff3',
+            'desc': "Test desc",
+            'due': None},
+            {'id': '456def',
+            'name': 'TestCard2',
+            'idList': '64fakedone999985b9aac33',
+            'desc': "Test desc 2",
+            'due': '2023-09-05T09:59:00.000Z'
+        }]
+        return StubResponse(fake_response_data)
+
+    raise Exception(f'Integration test did not expect URL "{url}"')
+
+# Stub replacement for requests.get(url)
+def stub_put_post(url, data={}):
+    test_card_id = "123abc"    
+    trello_key = os.environ.get('TRELLO_API_KEY')
+    trello_token = os.environ.get('TRELLO_TOKEN')
+
+    if url == f'https://api.trello.com/1/cards/{test_card_id}?key={trello_key}&token={trello_token}':
+        fake_response_data = [{
+            'id': '123abc',
+            'name': 'TestCard1',
+            'idList': '64fakedone999985b9aac33',
+            'desc': "Test desc",
+            'due': None
+        }]
+        return StubResponse(fake_response_data)
+    
+    elif url == f'https://api.trello.com/1/?key={trello_key}&token={trello_token}':
+        fake_response_data = [{
+            'id': '999aaa',
+            'name': 'TestAddCard2',
+            'idList': '64faketodo1234e86c4c2ff3',
+            'desc': "Test desc 3",
+            'due': None
+        }]
+        return StubResponse(fake_response_data)
+
+    raise Exception(f'Integration test did not expect URL "{url}"')
+
+def test_index_page(monkeypatch, client):
+    # This replaces any call to requests.get with our own function
+    monkeypatch.setattr(requests, 'get', stub)
+
+    response = client.get('/')
+
+    assert response.status_code == 200
+    assert 'TestCard1' in response.data.decode()
+    assert 'Test desc 2' in response.data.decode()
+    assert False == response.is_json
```

Comment:
Minor, but this feels like an odd assertion to me - do we really care that it's just "not json"? Would this encourage us to also check that it's "not XML", "not CSV" etc.?

Possibly flipping it on its head would more clearly spell out what you're trying to say, and looking at what this does [under the hood](https://werkzeug.palletsprojects.com/en/2.0.x/wrappers/#werkzeug.wrappers.Request.is_json) it's looking at the ContentType so maybe something like:
```python
assert "text/html" in response.content_type
```